capsules/src/process_console.rs

```diff
@@ -170,6 +174,68 @@ impl Default for WriterState {
     }
 }
 
+#[derive(Copy, Clone)]
+pub struct Command<const LEN: usize> {
+    buf: [u8; LEN],
+    len: usize,
+}
+impl<const LEN: usize> Command<LEN> {
+    pub fn new() -> Command<LEN> {
+        Command {
+            buf: [0; LEN],
+            len: 0,
+        }
+    }
+
+    pub fn fill(&mut self, buf: &[u8], terminator_idx: usize) -> Result<(), ErrorCode> {
+        if terminator_idx >= LEN {
+            self.len = LEN;
+        } else {
+            self.len = terminator_idx;
+        }
+
+        for i in 0..self.len {
+            self.buf[i] = buf[i];
+        }
+
+        Ok(())
+    }
+
+    pub fn is_buffer_empty(&mut self) -> Option<bool> {
+        Some(self.len == 0 && self.buf[0] == 0)
+    }
```

Comment:
Why do you need `Option<bool>` and not `bool`?

Reply:
Indeed is not needed, I though to include Option in order to match the rest of the structure methods like get_byte or is byte_empty, I will remove it